vertx-core/src/test/java/io/vertx/tests/http/WebSocketTest.java

```diff
@@ -3975,4 +4022,33 @@ public void testServerShutdownOverride() throws Exception {
       }));
     await();
   }
+
+  @Test
+  public void testCustomResponseHeadersBeforeUpgrade() throws InterruptedException {
+    String path = "/some/path";
+    String message = "here is some text data";
+    String headerKey = "custom";
+    String headerValue = "value";
+    server = vertx.createHttpServer(new HttpServerOptions().setPort(DEFAULT_HTTP_PORT)).requestHandler(req -> {
+      req.response().headers().set(headerKey, headerValue);
+      req.toWebSocket()
+        .onComplete(event -> {
```

Comment:
use `onSuccess`static method to wrap the block in order to ensure it works

Reply:
I've done the change.